src/hashcash.cr

```diff
@@ -11,131 +11,48 @@ module Hashcash
   # => 1:16:201206222403:hello:goodbyye:kfwaGRadlD3ddc9G:MTMxMzY5NQ==
   def self.generate(
     resource : String,
-    version = 1,
+    version = "1",
     bits = 20,
     date = Time.utc,
     ext = ""
-  )
-    Hashcash::Stamp.new(resource, version, bits, date, ext).generate
+  ) : String
+    hc = Hashcash::Stamp.new(resource, version, bits, date, ext)
+    hc.update_counter
+    hc.to_s
   end
 
   # Hashcash.verify?("1:20:201206222555:resource::pOWgc88+uDuefr/o:MTMxNzg2MA==", "resource")
   # => true
   # Hashcash.verify?("invalid_string", "resource")
   # => false
-  def self.verify?(hashcash_stamp : String, resource : String, time_window = 2.days.ago..2.days.from_now, bits = 20) : Bool
-    Hashcash::Stamp.new(resource).valid?(hashcash_stamp, resource, time_window, bits)
+  def self.verify?(
+    stamp_string : String,
+    resource : String,
+    time_window = 2.days.ago..2.days.from_now,
+    bits = 20
```

Comment:
So that defaults are guaranteed consistent, you may want to define `Hashcash::Stamp::DEFAULT_BITS = 20` and `Hashcash::Stamp::DEFAULT_TIME_WINDOW = 2.days`. These can then be used in all of these parameter defaults.